packages/provider/src/transports/base-provider-transport.ts

```diff
@@ -37,7 +37,7 @@ export abstract class BaseProviderTransport implements ProviderTransport {
 
   protected openPayload: { sessionId?: string; session?: WalletSession } | undefined
   protected connectPayload: ConnectDetails | undefined
-  protected accountsChangedPayload: string | undefined
+  protected accountsChangedPayload: { accounts: string[]; origin?: string } | undefined
```

Comment:
good catch, I also checked EIP1193, and this looks like the correct format to return (for `{accounts: []string}` and since origin is optional + additive, it wont break anything)